docs/rest_api.rst

```diff
@@ -0,0 +1,951 @@
+        "result": {}
+    }
+
+As before meta data can be chosen using *Rison* arguments::
+
+    (keys:!(label_columns))
+
+Will only fetch the *label_columns* meta data key
+
+And we can chose which columns to fetch::
+
+    (columns:!(name,address))
+
+To reduce or extend the default inferred columns from our *Model*.
+On server side we can use the ``list_columns`` property,
+this takes precedence over *Rison* arguments::
+
+    class ContactModelRestApi(ModelRestApi):
+        resource_name = 'contact'
+        datamodel = SQLAInterface(Contact)
+        list_columns = ['name', 'address']
+
+For ordering the results, the following will order contacts by name descending Z..A::
+
+    (order_column:name,order_direction:desc)
+
+To set a default order server side use ``base_order`` tuple::
+
+    class ContactModelRestApi(ModelRestApi):
+        resource_name = 'contact'
+        datamodel = SQLAInterface(Contact)
+        base_order = ('name', 'desc')
+
+Pagination, get the second page using page size of two (just an example)::
+
+    (page:2,page_size:2)
+
+To set the default page size server side::
+
+    class ContactModelRestApi(ModelRestApi):
+        resource_name = 'contact'
+        datamodel = SQLAInterface(Contact)
+        page_size = 20
+
+And last, but not least, *filters*. The query *filters* data structure::
+
+    {
+        "filters": [
+            {
+                "col": <COL_NAME>,
+                "opr": <Operation type>,
+                "value": <VALUE>
+            },
+            ...
+        ]
+    }
+
+All filters are **AND** operations. We can filter by several column names
+using different operations, so using *Rison*::
+
+    (filters:!((col:name,opr:sw,value:a),(col:name,opr:ew,value:z)))
+
+The previous filter will query all contacts whose **name** starts with "a" and ends with "z".
+The possible operations for each field can be obtained from the information endpoint.
+FAB can filter your models by any field type and all possible operations
+
+Note that all *Rison* arguments can be used alone or in combination::
+
+    (filters:!((col:name,opr:sw,value:a),(col:name,opr:ew,value:z)),columns:!(name),order_columns:name,order_direction:desc)
+
+Will filter all contacts whose **name** starts with "a" and ends with "z", using descending name order by, and
+just fetching the **name** column.
+
+To impose base filters server side::
+
+    class ContactModelRestApi(ModelRestApi):
+        resource_name = 'contact'
+        datamodel = SQLAInterface(Contact)
+        base_filters = [['name', FilterStartsWith, 'A']]
+
+The filter will act on all HTTP endpoints, protecting delete, create, update and display
+operations
+
+Simple example using doted notation, FAB will infer the necessary join operation::
+
+    class ContactModelRestApi(ModelRestApi):
+        resource_name = 'contact'
+        datamodel = SQLAInterface(Contact)
+        base_filters = [['contact_group.name', FilterStartsWith, 'F']]
```

Comment:
oh cool, so this supports other derivatives of `flask_appbuilder.models.sqla.filters` too right? Meaning we can reuse the same classes as we use against `ModelView`

Reply:
It's using the same ground, but without the MVC. Yes, I'm aiming for enabling an easy migration path.

Reply:
Yes the interface of writing a method that receives a SQLAlchemy query object and returns a new one (presumably with filters applied based on user attributes) works great for us.

Reply:
exactly!